tests/behat/context/ApiContext.php

```diff
@@ -3069,6 +3351,159 @@ public function iUploadAStandardCatrobatProgram(): void
+      },
+      'description' => function ($description)
+      {
+        Assert::assertIsString($description);
+      },
+      'version' => function ($version)
+      {
+        Assert::assertIsString($version);
+        Assert::assertRegExp('/[0-9]\\.[0-9]\\.[0-9]/', $version);
+      },
+      'views' => function ($views)
+      {
+        Assert::assertIsInt($views);
+      },
+      'download' => function ($download)
+      {
+        Assert::assertIsInt($download);
+      },
+      'private' => function ($private)
+      {
+        Assert::assertIsBool($private);
+      },
+      'flavor' => function ($flavor)
+      {
+        Assert::assertIsString($flavor);
+      },
+      'uploaded' => function ($uploaded)
+      {
+        Assert::assertIsInt($uploaded);
+      },
+      'uploaded_string' => function ($uploaded_string)
+      {
+        Assert::assertIsString($uploaded_string);
+      },
+      'screenshot_large' => function ($screenshot_large)
+      {
+        Assert::assertIsString($screenshot_large);
+        Assert::assertRegExp('/http:\\/\\/localhost\\/((resources_test\\/screenshots\/screen_[0-9]+)|(images\\/default\\/screenshot))\\.png/',
+          $screenshot_large);
+      },
+      'screenshot_small' => function ($screenshot_small)
+      {
+        Assert::assertIsString($screenshot_small);
+        Assert::assertRegExp('/http:\\/\\/localhost\\/((resources_test\\/thumbnails\/screen_[0-9]+)|(images\\/default\\/thumbnail))\\.png/',
+          $screenshot_small);
+      },
+      'project_url' => function ($project_url)
+      {
+        Assert::assertIsString($project_url);
+        Assert::assertRegExp('/http:\\/\\/localhost\\/app\\/project\\/[a-zA-Z0-9-]+/', $project_url);
+      },
+      'download_url' => function ($download_url)
+      {
+        Assert::assertIsString($download_url);
+        Assert::assertRegExp('/http:\\/\\/localhost\\/app\\/download\\/([a-zA-Z0-9-]+)\\.catrobat/',
+          $download_url);
+      },
+      'filesize' => function ($filesize)
+      {
+        Assert::assertEquals(is_float($filesize) || is_int($filesize), true);
+      },
+    ];
+    call_user_func($fields[$key], $program[$key]);
+
+    return true;
+  }
+
+  private function checkFeaturedProjectFieldsValue(array $program, string $key): bool
+  {
+    $fields = [
+      'id' => function ($id)
+      {
+        Assert::assertIsString($id);
+        Assert::assertRegExp('/^[a-zA-Z0-9-]+$/', $id, 'id');
+      },
+      'name' => function ($name)
+      {
+        Assert::assertIsString($name);
+      },
+      'author' => function ($author)
+      {
+        Assert::assertIsString($author);
+      },
+      'featured_image' => function ($featured_image)
+      {
+        Assert::assertIsString($featured_image);
+        Assert::assertRegExp('/http:\/\/localhost\/resources_test\/featured\/featured_[0-9]+\.jpg/',
+          $featured_image);
+      },
+    ];
+    call_user_func($fields[$key], $program[$key]);
```

Comment:
Have you tried out what happens if the method is called with a key that does not exist? I'm interested.